live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/utils/DockerTags.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.utils;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Optional;
+import java.util.Properties;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.SneakyThrows;
+
+/**
+ * Allows to access properties from 'docker-tags.properties' file. It's required to run: `mvn package` before accessing the properties file, to allow Maven filter out and replace variables.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class DockerTags
+{
+    private static final String PROPERTIES_FILE = "docker-tags.properties";
+    private static final String PROPERTIES_FILE_PATH = "target/test-classes/" + PROPERTIES_FILE;
+
+    private static Properties properties;
+
+    public static String getProperty(String key)
+    {
+        if (properties == null)
+        {
+            loadProperties();
+        }
+
+        return Optional.ofNullable(properties).map(p -> p.getProperty(key)).orElse(null);
+    }
+
+    public static String getOrDefault(String propertyKey, String defaultValue)
+    {
+        if (properties == null)
+        {
+            loadProperties();
```

Comment:
I am wondering if maybe we could use just `getProperty()` that will throw exception if property is not found and also make `loadProperties()` throw an error if properties doesn't exists as we probably expect that it should always be available and this approach will allow us to fail fast

Reply:
This is true, but I must admit, that it's quite annoying when you open a project, try to run a test and it fails because you need to make some step (e.g. `mvn package`) earlier. I like tests which run smoothly without any prerequisites. This is the reason why there are two methods: `getProperty()` and `getOrDefault()`, first one in did should fail if file is missing, but the second one shouldn't care if the file or property exists - if not return default value.
In the other hand there is this small risk, that even on CI something could go wrong with Docker image tag substitution because of what default value would be picked up and we wouldn't notice it. At the same time it's hard for me to image a situation where Maven would not handle the resource filtering correctly while building the app - something would have to be really messed up in config... and even if, would it be so bad to use the default value instead (even outdated)? We need to choose between convenience and reliability - I like when something works straight away, but in this case - having only one prerequisite - I could live with that. Tell me guys @Pawel-608 @tpage-alfresco @mpichura WDYT

Reply:
As we discussed we could:
* Do what you've done for all our "real" integration tests
* Add one test that only runs on CI which checks there are no `@something@` properties

We want to add a dev build profile anyway as there are some things that take a while and we don't need to do for each local build (e.g. download the license texts).